src/common/test/componentMocking.tsx

```diff
@@ -232,6 +354,74 @@ export const exposeProfileContext = (
   return { ...result, waitForDataChange, waitForUpdate, waitForErrorChange };
 };
 
+export const createResultPropertyTracker = <T,>({
+  renderHookResult,
+  valuePicker,
+}: {
+  renderHookResult: RenderHookResult<RenderHookResultsChildren, T>;
+  valuePicker: (props: T) => string | undefined | AnyObject;
+}): [() => Promise<void>] => {
+  const currentPicker = (): T => renderHookResult.result.current;
+  const waitForChange = () => {
+    const initialValue = valuePicker(currentPicker());
+    return new Promise<void>(async resolve => {
+      await waitFor(() => {
+        const newValue = valuePicker(currentPicker());
+        if (newValue !== initialValue) {
+          resolve();
+        } else {
+          throw new Error('waiting...');
+        }
+      });
+    });
+  };
+
+  return [waitForChange];
+};
+
+export const exposeProfileMutationsHook = (
+  responseProvider: ResponseProvider,
+  dataType: EditDataType
+): RenderHookResult<RenderHookResultsChildren, MutationReturnType> => {
+  const wrapper = ({ children }: RenderHookResultsChildren) => (
+    <MockApolloClientProvider responseProvider={responseProvider}>
+      <ProfileProvider>
+        <ProfileContextFetcher>{children}</ProfileContextFetcher>
+      </ProfileProvider>
+    </MockApolloClientProvider>
+  );
+
+  const callback = () =>
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useProfileMutations({
+      dataType,
+    });
+
+  return renderHook(callback, { wrapper });
+};
+
+export const ProfileContextFetcher = ({
+  children,
+}: {
+  children: React.ReactElement | React.ReactNodeArray;
+}): React.ReactElement => {
+  const [fetchStarted, setFetchStarted] = useState(false);
+  const { data, fetch } = useContext(ProfileContext);
+  if (!fetchStarted) {
+    fetch();
```

Comment:
Data fetching side effects are usually wrapped inside useEffect hook. Is it not needed this time?

Reply:
The idea of this component is to start fetching and only once. If useEffect was used, "fetch" must be hook's dependency and I can't guarantee if the Apollo's fetch function never changes between renders. Change would re-run the useEffect and fetch.